packages/xo-server/src/index.js

```diff
@@ -156,20 +212,15 @@ async function setUpPassport(express, xo) {
           return res.redirect('/signin')
         }
 
-        // The cookie will be set in via the next request because some
-        // browsers do not save cookies on redirect.
-        req.flash(
-          'token',
-          (await xo.createAuthenticationToken({ userId: user.id })).id
-        )
-
-        // The session is only persistent for internal provider and if 'Remember me' box is checked
-        req.flash(
-          'session-is-persistent',
+        req.session.user = { id: user.id, preferences: user.preferences }
+        req.session.persistent =
```

Comment:
I believe this two values are only necessary when using OTP.

Reply:
No, because we have move set token in new method who use `req.session` 
https://github.com/vatesfr/xen-orchestra/blob/2f5f33b94f0458b9e351604cd2f4162de7a5d271/packages/xo-server/src/index.js#L184

Reply:
Ok, then `setToken` should also remove the values when no longer necessary :)